crowbar_framework/app/models/service_object.rb

```diff
@@ -257,20 +257,31 @@ def remove_pending_elements(bc, inst, elements)
     end
   end
 
+  def restore_node_to_ready(node)
+    node.crowbar['state'] = 'ready'
+    node.crowbar['state_owner'] = ""
+    node.save
+  end
+
   def restore_to_ready(nodes)
     with_lock "BA-LOCK" do
       nodes.each do |node_name|
         node = NodeObject.find_node_by_name(node_name)
-        next if node.nil?
-
-        # Nothing to delay so mark them applying.
-        node.crowbar['state'] = 'ready'
-        node.crowbar['state_owner'] = ""
-        node.save
+        next if node.nil? || node.crowbar['state'] == 'crowbar-upgrade'
+          # Nothing to delay so mark them applying.
+          restore_node_to_ready(node)
       end
     end
   end
 
+  def remove_upgrade_state(nodes)
+    nodes.each do |node_name|
+      node = NodeObject.find_node_by_name(node_name)
+      next if node.nil?
+        restore_node_to_ready(node)
+    end
+  end
+
```

Comment:
I am pretty sure with the current state of the updater PR none of the changes here are really needed.